### ISSUE TITLE
idea: make code importable with ES Modules.

### ISSUE THREAD

**Issue by trusktr** [felixmariotto/three-mesh-ui#9]
Alternatively, we can update the webpack build to output a single ESM file like `build/three-mesh-ui.esm.js`, and also publish that to NPM.

---

Alternatively I think we could add `.js` to all the module identifiers inside of `src/`.

Then we could import from src directly:

```js
import('https://unpkg.com/three-mesh-ui@1.3.0/src/three-mesh-ui.js').then(console.log)
```

In Chrome you should currently get an error like

```
Uncaught (in promise) TypeError: Failed to resolve module specifier "three". Relative references must start with either "/", "./", or "../".
```

Besides changing import specifiers to use `.js`, we would also need to release an importmap so that it knows how to resolve three (we can have it point to `https://unpkg.com/three@0.115.0`).

Maybe the first option is easier, but requires a build step.

**Comment by felixmariotto** [comment]
TBH I'm a bit lost in all these ways of importing libraries, if you have an idea on how to improve this feel free to file a PR !

**Comment by trusktr** [comment]
> TBH I'm a bit lost in all these ways of importing libraries

Yeah. It'd been a mess! Things are starting to stabilize now that Node.js has ESM. It'll be a while before things converge better.

**Comment by felixmariotto** [comment]
Yeah, I don't know... When I hear about deno.js that doesn't support NPM...

**Comment by trusktr** [comment]
Deno is a whole 'nother beast. I think this weekend I'll start with this issue first, to help me get more familiar with everything. :)

**Comment by trusktr** [comment]
Welp! ^ "this weekend" was a bad estimate, but I ended up starting with https://github.com/felixmariotto/three-mesh-ui/pull/31 which helped get more familiar with stuff.

**Comment by felixmariotto** [comment]
No problem, anyway as you probably have seen the lib structure wildly changed at its beginning. It's finally in a more mature and stable state, so it makes more sense to contribute now.   
I'm actually working on games now, which I needed three-mesh-ui for.
